cmd/autoheal/object_template.go

```diff
@@ -121,33 +121,55 @@ func (t *ObjectTemplate) Process(object interface{}, data interface{}) error {
 	if glog.V(2) {
 		glog.Infof("Data: %v", data)
 	}
-	return t.processValue(reflect.ValueOf(object), data)
+	v, err := t.processValue(reflect.ValueOf(object), data)
+	if err == nil {
+		object = v.Interface()
```

Comment:
Are you sure?
From what I tried this seems to work.
How else can I make sure `object` is updated?

Reply:
In this context `object` is a local variable of the `Process` method. If you assign something to it you are just changing the local variable, which won't have any effect in the original one, because it is passed by value. If you want to return a new result then you will have to use a return value, or else pass a pointer to the interface:

```go
func (t *ObjectTemplate) Process(input interface{}, data interface{}) (output interface{}, err) {
  ...
  output = ...
}
```

Or, with a pointer:

```go
func (t *ObjectTemplate) Process(object *interface{}, data interface{}) error {
  ...
  *object = ...
}
```

The second alternative is very confusing, so I'd suggest the first.

Reply:
Yes, I think that adding that check is enough.

Reply:
As you added the check, I think you can now remove this assignment.